File: Backend/controllers/locationController.js
const Lokasi = require('../models/locationModel');

const LocationController = {
  getAll: async (req, res) => {
    try {
      const data = await Lokasi.findAll();
      res.json({ success: true, data });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  },

  getById: async (req, res) => {
    try {
      const data = await Lokasi.findById(req.params.id);
      if (!data) return res.status(404).json({ success: false, message: 'Lokasi tidak ditemukan' });
      res.json({ success: true, data });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  },

  create: async (req, res) => {
    try {
      if (!req.body.nama_lokasi || !req.body.nama_lokasi.trim()) {
        return res.status(400).json({ success: false, message: 'Nama lokasi harus diisi' });
      }
      const id = await Lokasi.create({ ...req.body, user_id: req.user.id });
      const newData = await Lokasi.findById(id);
      res.status(201).json({ success: true, data: newData });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  },

  update: async (req, res) => {
    try {
      if (!req.body.nama_lokasi || !req.body.nama_lokasi.trim()) {
        return res.status(400).json({ success: false, message: 'Nama lokasi harus diisi' });
      }
      const affected = await Lokasi.update(req.params.id, req.body);
      if (!affected) return res.status(404).json({ success: false, message: 'Lokasi tidak ditemukan' });
      const updated = await Lokasi.findById(req.params.id);
      res.json({ success: true, data: updated });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  },

  delete: async (req, res) => {
    try {
      const affected = await Lokasi.delete(req.params.id);
      if (!affected) return res.status(404).json({ success: false, message: 'Lokasi tidak ditemukan' });
      res.json({ success: true, message: 'Lokasi dihapus' });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  }
};

module.exports = LocationController;

File: backend_python/controllers/location_controller.py
from flask import g, request

from models.location_model import Lokasi
from utils.helpers import fail, ok


class LocationController:
    @staticmethod
    def get_all():
        return ok({"success": True, "data": Lokasi.find_all()})

    @staticmethod
    def get_by_id(row_id):
        row = Lokasi.find_by_id(row_id)
        if not row:
            return fail("Lokasi tidak ditemukan", 404)
        return ok({"success": True, "data": row})

    @staticmethod
    def create():
        body = request.get_json(silent=True) or {}
        if not body.get("nama_lokasi") or not str(body["nama_lokasi"]).strip():
            return fail("Nama lokasi harus diisi", 400)
        new_id = Lokasi.create({**body, "user_id": g.user["id"]})
        return ok({"success": True, "data": Lokasi.find_by_id(new_id)}, 201)

    @staticmethod
    def update(row_id):
        body = request.get_json(silent=True) or {}
        if not body.get("nama_lokasi") or not str(body["nama_lokasi"]).strip():
            return fail("Nama lokasi harus diisi", 400)
        affected = Lokasi.update(row_id, body)
        if not affected:
            return fail("Lokasi tidak ditemukan", 404)
        return ok({"success": True, "data": Lokasi.find_by_id(row_id)})

    @staticmethod
    def delete_row(row_id):
        affected = Lokasi.delete(row_id)
        if not affected:
            return fail("Lokasi tidak ditemukan", 404)
        return ok({"success": True, "message": "Lokasi dihapus"})
